optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/view/IndictmentView.java

```diff
@@ -0,0 +1,100 @@
+package org.optaplanner.openshift.employeerostering.shared.roster.view;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
+import org.optaplanner.core.api.score.constraint.ConstraintMatch;
+import org.optaplanner.core.api.score.constraint.Indictment;
+import org.optaplanner.persistence.jackson.api.score.ScoreJacksonJsonSerializer;
+import org.optaplanner.persistence.jackson.api.score.buildin.hardsoft.HardSoftScoreJacksonJsonDeserializer;
+
+public class IndictmentView {
+
+    private List<Constraint> constraintMatchedList;
+    private List<HardSoftScore> scoreList;
+
+    public IndictmentView() {
+
+    }
+
+    public IndictmentView(Indictment indictment) {
+        constraintMatchedList = new ArrayList<>(indictment.getConstraintMatchCount());
+        scoreList = new ArrayList<>(indictment.getConstraintMatchCount());
+
+        for (ConstraintMatch cm : indictment.getConstraintMatchSet()) {
+            constraintMatchedList.add(Constraint.forConstraintMatch(cm));
+            scoreList.add((HardSoftScore) cm.getScore());
+        }
+    }
+
+    public IndictmentView(List<Constraint> ruleMatched, List<HardSoftScore> score) {
+        this.constraintMatchedList = ruleMatched;
+        this.scoreList = score;
+    }
+
+    public List<Constraint> getConstraintMatchedList() {
+        return constraintMatchedList;
+    }
+
+    @JsonSerialize(contentUsing = ScoreJacksonJsonSerializer.class)
+    @JsonDeserialize(contentUsing = HardSoftScoreJacksonJsonDeserializer.class)
+    public List<HardSoftScore> getScoreList() {
+        return scoreList;
+    }
+
+    public void setConstraintMatchedList(List<Constraint> constraintMatchedList) {
+        this.constraintMatchedList = constraintMatchedList;
+    }
+
+    public void setScoreList(List<HardSoftScore> scoreList) {
+        this.scoreList = scoreList;
+    }
+
+    public static enum Constraint {
```

Comment:
Let's rename this to RosterConstraintType and put it as a normal class (not an inner class) in this shared.roster.view package. I suspect we'll need to this more later, once we do the Parameterization feature.

Reply:
Done